lib/mayaUsd/ufe/MayaUsdUIInfoHandler.cpp

```diff
@@ -88,5 +99,68 @@ UsdUfe::UsdUIInfoHandler::SupportedTypesMap MayaUsdUIInfoHandler::getSupportedIc
     return supportedTypes;
 }
 
+Ufe::UIInfoHandler::Icon
+MayaUsdUIInfoHandler::treeViewIcon(const Ufe::SceneItem::Ptr& mayaItem) const
+{
+    Ufe::UIInfoHandler::Icon icon = Parent::treeViewIcon(mayaItem);
+
+    if (icon.baseIcon == "out_USD_Shader.png") {
+        // Naming convention for third party shader outliner icons:
+        //
+        //  We take the info:id of the shader and make it safe by replacing : with _.
+        //  Then we search the Maya icon paths for a PNG file with that name. If found we will use
+        //  it. Please note that files with _150 and _200 can also be provided for high DPI
+        //  displays.
+        //
+        //   For example an info:id of:
+        //       MyRenderer:nifty_surface
+        //   Will have this code search the full Maya icon path for a file named:
+        //       MyRenderer_nifty_surface.png
```

Comment:
That convention was the result of a discussion with our renderer team for a universal convention that also included:
- Native MaterialX nodes in the outliner
- Scalable SVG icons for LookdevX in both USD and MaterialX runtimes

I can see the usefulness of the `out_USD_` prefix for shorter names which might exist somewhere else on the Maya icon path. In the case of MaterialX nodes, we will see the characteristic `ND_` typical of MaterialX nodes, and in the case of Arnold nodes they will always be prefixed with the renderer name. I hope this will be sufficient to prevent name clashes.

We can always re-open the naming convention to add a single universal prefix if you think the risk of collisions is too great.

Reply:
Its not just about name clashes. The icons for Outliner are prefixed with "out_", and since these are coming via Ufe, we also add the ufe runtime name, so "out_USD_". Can you ask the renderer team if they are okay with not following the usual icon naming scheme in favor of a universal convention.